smart_contracts.rst

```diff
@@ -762,9 +762,10 @@ Also, the amount that a participant can receive cannot be bigger than the total
 
 ::
 
-    (7 R) T2 + L2 - T1 - L1 <= D2
+    (7 R) -D1 <= T2 + L2 - T1 - L1 <= D2
```

Comment:
btw, this is missing the withdraws:

`-(D1 - W1) <= T2 + L2 - T1 - L1 <= D2 - W2`

Reply:
Are you sure? `T + L` are the total transferred amounts. The netted total transferred amounts should be bounded by the channel deposit. 
The withdraws are then taken from the available balance, but we do not subtract them from the total transferred amounts.

Reply:
> The netted total transferred amounts should be bounded by the channel deposit.

Nope, they are bound by the participant's capacity, which is D - W.

You cannot have a net value that is larger then your available deposit on chain, consider:

P2 Deposit 10
P2 Withdraws 10
If P1 enforce just `T2 + L2 - T1 - L1 <= D2` it means P2 can make transfer up to 10 tokens, which is wrong.

Additionally, since you're talking about the off-chain client, `D2 - W2` is **not** sufficient, since `W2` is defined as the current on-chain withdraw. The client must enforce this against the **off**-chain withdraw, which is different from the on-chain value.

Reply:
I think you did not consider `P1`'s transferred amounts. `P2` can transfer up to `10` tokens if `P1` owes him `10` tokens.

E.g.:

```
D1 = 10
W1 = 0
T1 = 4
L1 = 0


D2 = 10
W2 = 5
T2 = 10
L2 = 0
```

```sh
T2 + L2 - T1 - L1 = 10 + 0 - 4 - 0 = 6
# The following is true:
# - D1 <= 6 <= D2
# - 10 <= 6 <= 10

# The following is not true:
# -(D1 - W1) <= 6 <= D2 - W2
# - (10 - 0) <= 6 <= 10 - 5
# -10 <= 6 <= 5
# 6 <= 5 not true
```